### PR TITLE
features: promote NodeLogQuery to beta

### PR DESCRIPTION
**Pull request by aravindhp**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Promote NodeLogQuery feature to beta

#### Special notes for your reviewer:
PRR review:
- https://github.com/kubernetes/enhancements/pull/4480
#### Does this PR introduce a user-facing change?
```release-note
Promote the NodeLogQuery feature to beta. There have been no functional changes from alpha.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/2258-node-log-query
- https://github.com/kubernetes/website/pull/45126
```

### CODE REVIEW

**Comment by bart0sh** [comment]
/triage accepted
/priority important-soon

@aravindhp Are there any documentation updates related to this? Is this promotion included into the [SIG-Node planning doc](https://docs.google.com/document/d/1U10J0WwgWXkdYrqWGGvO8iH2HKeerQAlygnqgDgWv4E/edit#heading=h.lmmz2ofc5ko)?

**Comment by aravindhp** [comment]
> /triage accepted /priority important-soon

Thank you

> @aravindhp Are there any documentation updates related to this? 

The only doc update is changing the feature state in https://kubernetes.io/docs/concepts/cluster-administration/system-logs/#log-query. I was going to open that PR after this merges.

>Is this promotion included into the [SIG-Node planning doc](https://docs.google.com/document/d/1U10J0WwgWXkdYrqWGGvO8iH2HKeerQAlygnqgDgWv4E/edit#heading=h.lmmz2ofc5ko)?

It is not. Should I add an item here?

**Comment by bart0sh** [comment]
> The only doc update is changing the feature state in https://kubernetes.io/docs/concepts/cluster-administration/system-logs/#log-query. I was going to open that PR after this merges.

Sounds good, thanks for the explanations. 

> It is not. Should I add an item there?

I don't think it's needed as the owning sig is sig-windows. I thought it's SIG-Node KEP.

**Comment by kannon92** [comment]
Looking at the KEP, you mention that these e2e tests are covered by https://testgrid.k8s.io/sig-windows-master-release#capz-master-windows-alpha-nodelogquery?

When you promote this to beta, do you have an existing job you can confirm hits these e2e tests?

**Comment by aravindhp** [comment]
@kannon92 I have setup a [daily periodic job](https://github.com/kubernetes/test-infra/blob/393f4ee2f7a66a7277c641667db5dba17a3ce12c/config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml#L491) to test this.

**Comment by kannon92** [comment]
> @kannon92 I have setup a [daily periodic job](https://github.com/kubernetes/test-infra/blob/393f4ee2f7a66a7277c641667db5dba17a3ce12c/config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml#L491) to test this.

I see. The test is labelled alpha but you are really only testing `NodeLogQuery` with it. I think thats fine. I was worried you were promoting this to beta and that test would no longer apply here.

**Comment by kannon92** [comment]
/test pull-kubernetes-e2e-capz-windows-alpha-features
It'd be nice to have a presubmit to verify that this feature is working. If there isn't a corresponding presubmit for the periodic, that may be helpful to add.

**Comment by aravindhp** [comment]
> I see. The test is labelled alpha but you are really only testing `NodeLogQuery` with it. I think thats fine. I was worried you were promoting this to beta and that test would no longer apply here.

Yes I am only testing `NodeLogQuery` with it. I was planning on renaming the test after the feature graduates to beta. I did not want to rename it before it graduated.

> It'd be nice to have a presubmit to verify that this feature is working. If there isn't a corresponding presubmit for the periodic, that may be helpful to add.

I can look into doing that. I was not sure about doing that given this feature will always default to off even when it hits GA. But for beta the periodic does verify that this feature is working.

**Comment by kannon92** [comment]
Yea, you can take my comments as non-blocking. You can only really confirm this test works by merging this change and monitoring the periodic. Thats fine but not ideal. Since you have a signal for the feature that you created I'm fine with what you have.

I just notice sometimes its nice to have a trigger on the PR so you can confirm that this feature is working as you want while you work on PRs.

**Comment by rphillips** [comment]
Tests were added in:

* https://github.com/kubernetes/kubernetes/pull/122240
* https://github.com/kubernetes/kubernetes/pull/122594

/lgtm

**Review by mrunalp — APPROVED** [review]

**Comment by aravindhp** [comment]
> Is there also a docs update for this promotion (usually we'd expect one)?

@sftim. There is:
- https://github.com/kubernetes/website/pull/45126
I thought I needed to wait for this PR to merge before opening the doc PR. I have updated the description and added a release note to the PR. Apologies for missing this initially.

**Pull request merged**
